runtimes/runtimes/lsp/router/lspServer.ts

```
@@ -87,22 +87,29 @@ export class LspServer {
         params: InitializeParams,
         token: CancellationToken
     ): Promise<PartialInitializeResult | ResponseError<InitializeError> | undefined> => {
-        this.clientSupportsNotifications =
-            params.initializationOptions?.aws.awsClientCapabilities?.window?.notifications
+        try {
+            this.clientSupportsNotifications =
+                params.initializationOptions?.aws?.awsClientCapabilities?.window?.notifications
 
-        if (!this.initializeHandler) {
-            return
-        }
+            if (!this.initializeHandler) {
+                return
+            }
 
-        const initializeResult = await asPromise(this.initializeHandler(params, token))
-        if (!(initializeResult instanceof ResponseError)) {
-            this.initializeResult = initializeResult
-            if (initializeResult?.serverInfo) {
-                this.notificationRouter = new RouterByServerName(initializeResult.serverInfo.name, this.encoding)
+            const initializeResult = await asPromise(this.initializeHandler(params, token))
+            if (!(initializeResult instanceof ResponseError)) {
+                this.initializeResult = initializeResult
+                if (initializeResult?.serverInfo) {
+                    this.notificationRouter = new RouterByServerName(initializeResult.serverInfo.name, this.encoding)
+                }
             }
-        }
 
-        return initializeResult
+            return initializeResult
+        } catch (error) {
```

What's the reason to try/catch here?

---

I think with this change LSP server will start even if Flare servers inside failed to initialize, right? How will client know about that?

---

yes the primary reason was for logging but indeed, let's re-throw the exception.

---

fix in this PR: https://github.com/aws/language-server-runtimes/pull/307